docs/sources/flow/reference/components/otelcol.receiver.kafka.md

```diff
@@ -0,0 +1,304 @@
+
+### aws_msk block
+
+The `aws_msk` block configures extra parameters for SASL authentication when
+using the `AWS_MSK_IAM` mechanism.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`region` | `string` | AWS region the MSK cluster is based in. | | yes
+`broker_addr` | `string` | MSK address to connect to for authentication. | | yes
+
+### tls block
+
+The `tls` block configures TLS settings used for connecting to the Kafka
+brokers. If the `tls` block isn't provided, TLS won't be used for
+communication.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`ca_file` | `string` | Path to the CA file. | | no
+`cert_file` | `string` | Path to the TLS certificate. | | no
+`key_file` | `string` | Path to the TLS certificate key. | | no
+`min_version` | `string` | Minimum acceptable TLS version for connections. | `"TLS 1.2"` | no
+`max_version` | `string` | Maximum acceptable TLS version for connections. | `"TLS 1.3"` | no
+`reload_interval` | `duration` | Frequency to reload the certificates. | | no
+`client_ca_file` | `string` | Path to the CA file used to authenticate client certificates. | | no
+
+### kerberos block
+
+The `kerberos` block configures Kerberos authentication against the Kafka
+broker.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`service_name` | `string` | Kerberos service name. | | no
+`realm` | `string` | Kerberos realm. | | no
+`use_keytab` | `string` | Enables using keytab instead of password. | | no
+`username` | `string` | Kerberos username to authenticate as. | | yes
+`password` | `secret` | Kerberos password to authenticate with. | | no
+`config_file` | `string` | Path to Kerberos location (for example, `/etc/krb5.conf`). | | no
+`keytab_file` | `string` | Path to keytab file (for eaxmple, `/etc/security/kafka.keytab`). | | no
+
+When `use_keytab` is `false`, the `password` argument is required. When
+`use_keytab` is `true`, the file pointed to by the `keytab_file` argument is
+used for authentication instead. At most one of `password` or `keytab_file`
+must be provided.
+
+### metadata block
+
+The `metadata` block configures how to retrieve and store metadata from the
+Kafka broker.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`include_all_topics` | `bool` | When true, maintains metadata for all topics. | `true` | no
+
+If the `include_all_topics` argument is `true`, `otelcol.receiver.kafka`
+maintains a full set of metadata for all topics rather than the minimal set
+that has been necessary so far. Including the full set of metadata is more
+convenient for users, can consume a substantial amount of memory if you have
+many topics and partitions.
```

Comment:
```suggestion
If the `include_all_topics` argument is `true`, `otelcol.receiver.kafka`
maintains a full set of metadata for all topics rather than the minimal set
that has been necessary so far. Including the full set of metadata is more
convenient for users but can consume a substantial amount of memory if you have
many topics and partitions.
```